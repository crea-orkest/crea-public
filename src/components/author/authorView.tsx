import type { Author as AuthorType } from 'graphql/types/author'
import Image from 'next/image'
import React from 'react'
import classNames from 'classnames'
import styles from './styles.module.scss'

export interface Props {
  data: AuthorType
}

export const AuthorView: React.FC<Props> = ({ data }: Props) => {
  return (
    <div className={classNames(styles.author)}>
      <ul className={classNames(styles.author__metadata)}>
        <li className={classNames(styles.author__name)}>
          <strong>{data.name}</strong>
        </li>
        <li className={classNames(styles.author__description)}>
          {data.description}
        </li>
      </ul>
      {data.image?.url && (
        <Image
          className={classNames(styles.author__profilePicture)}
          src={data.image.url}
          alt={data.image.description}
          height={64}
          width={64}
        />
      )}
    </div>
  )
}
