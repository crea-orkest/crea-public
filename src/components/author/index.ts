export { AuthorContainer as Author } from './authorContainer'
export { AuthorView } from './authorView'
