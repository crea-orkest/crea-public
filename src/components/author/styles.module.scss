.author {
  display: grid;
  grid-template-columns: 4rem 10rem;
  grid-template-rows: 2rem 2rem;

  gap: 0 1rem;

  &__metadata {
    grid-column: 2;
    grid-row: 1 / span 2;
    align-self: center;

    list-style: none;
    padding: 0;
  }

  &__name {
  }

  &__description {
  }

  &__profilePicture {
    grid-column: 1 / span 1;
    grid-row: 1 / span 2;
    width: 4rem;
    height: 4rem;
    object-fit: cover;
    border-radius: 50%;
  }
}
