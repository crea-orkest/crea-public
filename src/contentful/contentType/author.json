{
  "sys": {
    "space": {
      "sys": { "type": "Link", "linkType": "Space", "id": "pwuxt9271wes" }
    },
    "id": "author",
    "type": "ContentType",
    "createdAt": "2023-06-26T18:59:32.061Z",
    "updatedAt": "2023-06-26T18:59:32.483Z",
    "environment": {
      "sys": { "id": "master", "type": "Link", "linkType": "Environment" }
    },
    "publishedVersion": 1,
    "publishedAt": "2023-06-26T18:59:32.483Z",
    "firstPublishedAt": "2023-06-26T18:59:32.483Z",
    "createdBy": {
      "sys": {
        "type": "Link",
        "linkType": "User",
        "id": "7dyBSKYa3yVW3xUX4g19qy"
      }
    },
    "updatedBy": {
      "sys": {
        "type": "Link",
        "linkType": "User",
        "id": "7dyBSKYa3yVW3xUX4g19qy"
      }
    },
    "publishedCounter": 1,
    "version": 2,
    "publishedBy": {
      "sys": {
        "type": "Link",
        "linkType": "User",
        "id": "7dyBSKYa3yVW3xUX4g19qy"
      }
    }
  },
  "displayField": "name",
  "name": "Author",
  "description": "",
  "fields": [
    {
      "id": "name",
      "name": "Name",
      "type": "Symbol",
      "localized": false,
      "required": true,
      "validations": [{ "unique": true }],
      "disabled": false,
      "omitted": false
    },
    {
      "id": "description",
      "name": "description",
      "type": "Text",
      "localized": true,
      "required": false,
      "validations": [],
      "disabled": false,
      "omitted": false
    },
    {
      "id": "image",
      "name": "image",
      "type": "Link",
      "localized": false,
      "required": false,
      "validations": [{ "linkMimetypeGroup": ["image"] }],
      "disabled": false,
      "omitted": false,
      "linkType": "Asset"
    }
  ]
}
