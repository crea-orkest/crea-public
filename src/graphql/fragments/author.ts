import { gql } from "@apollo/client";

export const author = gql`
  fragment author on PersonRecord {
    id
    name
    role
  }
`;
