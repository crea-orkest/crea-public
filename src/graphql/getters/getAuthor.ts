import type { RequestInit } from 'next/dist/server/web/spec-extension/request'
import { authorFormatter } from '../formatters/authorFormatter'
import { client } from '../gqlClient'
import {
  GetAuthorDocument,
  type GetAuthorQuery,
  type GetAuthorQueryVariables,
} from '../generated/graphql'

export const getAuthor = async (
  { id }: GetAuthorQueryVariables,
  fetchOptions?: RequestInit
) => {
  try {
    const { data, error } = await client.query<
      GetAuthorQuery,
      GetAuthorQueryVariables
    >(GetAuthorDocument, { id }, { fetchOptions })

    return {
      data: data ? authorFormatter(data) : null,
      error,
    }
  } catch (error) {
    if (error instanceof Error) console.log(error.message)

    return { data: null, error }
  }
}
