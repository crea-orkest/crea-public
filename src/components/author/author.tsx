import { getAuthor } from "graphql/getters/getAuthor";
import React from "react";

export interface Props {
  id: string | undefined;
}

export const Author: React.FC<Props> = async ({ id }) => {
  if (!id) return null;
  const { data } = await getAuthor({ id });
  if (!data) return null;

  return (
    <div>
      <p>{data.name}</p>
      <p>{data.description}</p>
    </div>
  );
};
