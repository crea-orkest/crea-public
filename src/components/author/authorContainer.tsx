import React from 'react'
import { AuthorView } from './authorView'
import { getAuthor } from '../../graphql/getters/getAuthor'

export interface Props {
  id: string
}

export const AuthorContainer: React.FC<Props> = async ({ id }) => {
  const { data } = await getAuthor({ id })
  if (!data) return null

  return <AuthorView data={data} />
}
