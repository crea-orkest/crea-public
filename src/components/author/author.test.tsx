import { Author } from "./author";
import type { Props } from "./author";
import React from "react";
import { getAuthor } from "graphql/getters/getAuthor";
import { resolvedComponent } from "../../testHelpers/resolvedComponent";
import { render, screen } from "@testing-library/react";

jest.mock("../../graphql/getters/getAuthor", () => {
  return {
    __esModule: true,
    getAuthor: jest.fn(),
  };
});

const getAuthorMock = jest.mocked(getAuthor);

describe("Author component", () => {
  it("shows all the data", async () => {
    getAuthorMock.mockResolvedValueOnce({
      data: {
        id: "id",
        name: "name",
        description: "role",
      },
      errors: undefined,
    });

    const Resolved = await resolvedComponent<Props>(Author, { id: "some id" });

    render(<Resolved />);

    expect(screen.getByText("name")).toBeTruthy();
    expect(screen.getByText("role")).toBeTruthy();
  });
});
