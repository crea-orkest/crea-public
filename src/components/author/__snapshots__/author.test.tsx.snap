// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author component shows all the data 1`] = `
<div>
  <div>
    <p>
      name
    </p>
    <p>
      role
    </p>
  </div>
</div>
`;
